core/dbt/parser/base.py

```diff
@@ -414,6 +417,22 @@ def parse_file(self, file_block: FileBlock) -> None:
         self.parse_node(file_block)
 
 
+# TOOD: Currently `ModelParser` inherits from `SimpleSQLParser`, which inherits from
+# SQLParser. So possibly `ModelParser` should instead be `SQLModelParser`, and in
+# `ManifestLoader.load`, we should resolve which to use? (I think that's how it works;
+# though then why all the inheritance and generics if every model is parsed with `ModelParser`?)
+# class PRQLParser(
+#     ConfiguredParser[FileBlock, IntermediateNode, FinalNode], Generic[IntermediateNode, FinalNode]
+# ):
```

Comment:
It sounds like this piece is the biggest current challenge, and the thing that would require the most refactoring to make pluggable.

Currently, `SimpleSQLParser` is used in all of these places:
- `ModelParser` (even for Python models)
- `AnalysisParser`
- `SingularTestParser`
- `SqlBlockParser` (for semantic layer queries)
- `SeedParser` (!), even though `render_with_context` is a total no-op

We shouldn't be looking to create new node types for each language — these should still be models, analyses, tests, etc — but it's clear that we need some way to allow a language plugin to define its own parser, and then provide it to resources matching that language input.

It also looks like the real Jinja-SQL entrypoint is `ConfiguredParser.render_with_context`. We'd probably want to split that out, to live on a dedicated `JinjaSQLParser` instead!

Caveats:
- There's some additional Jinja business in `update_parsed_node_config_dict` and `update_parsed_node_name`, to respect rules around hooks and database/schema/alias — those have more to do with model configuration than model language, and the fact that we've stretched Jinja into a "rules engine" for configuration that goes beyond its more-straightforwardly understood role as a SQL template.
- I'd like to punt on snapshots, which are defined _as_ Jinja blocks (and mostly config, anyway)
- I don't know if we should even think/talk about macros, at this early point, but we know we'll want _some_ way to define and reuse [functions](https://prql-lang.org/book/queries/functions.html)